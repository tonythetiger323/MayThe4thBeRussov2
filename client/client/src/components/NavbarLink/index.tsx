import * as React from 'react'
import { Link } from 'react-router-dom'
import Button, { ButtonProps } from '@material-ui/core/Button'
import { History } from 'history'

// https://material-ui.com/guides/typescript/#usage-of-component-property
// creating a specialized button component that has the props of
// react-router-dom link passed to it

interface NavbarLinkProps extends ButtonProps {
  readonly to: History.LocationDescriptor;
  readonly children: any;
}

const NavbarLink: React.SFC<NavbarLinkProps> = props => {
  const renderLink: React.SFC<any> = itemProps => (
    <Link to={props.to} {...itemProps} />
  );

  return (
    <Button component={renderLink} {...props}>
      {props.children}
    </Button>
  );
};
export default NavbarLink;
