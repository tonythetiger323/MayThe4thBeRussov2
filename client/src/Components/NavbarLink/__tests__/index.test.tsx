import * as React from 'react'
import { shallow } from 'enzyme'
import NavbarLink from '../../NavbarLink'

let wrapper: any

describe('NavbarLink Component', () => {
  beforeEach(
    () => (wrapper = shallow(<NavbarLink to={'/'} children={'props'} />))
  )

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
