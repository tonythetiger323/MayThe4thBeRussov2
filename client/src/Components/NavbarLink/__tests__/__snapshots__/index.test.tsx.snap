// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarLink Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavbarLink
    to="/"
  >
    props
  </NavbarLink>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "props",
      "component": [Function],
      "to": "/",
    },
    "ref": null,
    "rendered": "props",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "props",
        "component": [Function],
        "to": "/",
      },
      "ref": null,
      "rendered": "props",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
